lib/quantization/src/encoded_vectors_u8.rs

```diff
@@ -509,7 +509,22 @@ impl<TStorage: EncodedStorage> EncodedVectors for EncodedVectorsU8<TStorage> {
     }
 
     fn encode_internal_vector(&self, id: u32) -> Option<EncodedQueryU8> {
+        let offset_difference = match self.metadata.vector_parameters.distance_type {
+            DistanceType::Dot => {
+                self.metadata.actual_dim as f32 * self.metadata.offset * self.metadata.offset
+            }
+            DistanceType::L1 => 0.0,
+            DistanceType::L2 => {
+                self.metadata.actual_dim as f32 * self.metadata.offset * self.metadata.offset
+            }
```

Comment:
These are the same. We can merge them if we like:

```suggestion
            DistanceType::Dot | DistanceType::L2 => {
                self.metadata.actual_dim as f32 * self.metadata.offset * self.metadata.offset
            }
            DistanceType::L1 => 0.0,
```